metals/src/main/scala/scala/meta/internal/worksheets/WorkspaceEditWorksheetPublisher.scala

```diff
@@ -0,0 +1,168 @@
+package scala.meta.internal.worksheets
+
+import scala.meta.internal.metals.MetalsLanguageClient
+import mdoc.interfaces.EvaluatedWorksheet
+import scala.meta.io.AbsolutePath
+import org.eclipse.lsp4j.WorkspaceEdit
+import org.eclipse.lsp4j.ApplyWorkspaceEditParams
+import org.eclipse.lsp4j.TextEdit
+import org.eclipse.lsp4j.jsonrpc.messages.{Either => JEither}
+import org.eclipse.lsp4j.MarkedString
+import scala.meta.internal.metals.MetalsEnrichments._
+import mdoc.interfaces.EvaluatedWorksheetStatement
+import scala.meta.inputs.Input
+import org.eclipse.lsp4j.{Position, Range}
+import scala.meta.internal.metals.Buffers
+import org.eclipse.lsp4j.Hover
+import scala.meta.internal.metals.TokenEditDistance
+import WorkspaceEditWorksheetPublisher._
+
+class WorkspaceEditWorksheetPublisher(buffers: Buffers)
+    extends WorksheetPublisher {
+
+  private var hoverMessages = Map.empty[AbsolutePath, HoverMap]
+
+  override def publish(
+      languageClient: MetalsLanguageClient,
+      path: AbsolutePath,
+      worksheet: EvaluatedWorksheet
+  ): Unit = {
+    (render(path) _ andThen publish(path, languageClient))(worksheet)
```

Comment:
nit: let's avoid infix operations `a op b` for non-symbol operators. I'm not sure what the correct refactoring is but something like this below would be easier to read

```suggestion
    publish(path, languageClient, render(path, worksheet))
```

Reply:
done. Named `rendered` result and then passed it to `publish`